oidc-controller/api/routers/oidc.py

```diff
@@ -37,6 +39,18 @@ async def poll_pres_exch_complete(pid: str, db: Database = Depends(get_db)):
     """Called by authorize webpage to see if request
     is verified and token issuance can proceed."""
     auth_session = await AuthSessionCRUD(db).get(pid)
+
+    """
+     Check if proof is expired. But only if the proof has not been started.
+     NOTE: This should eventually be moved to a background task.
+    """
+    if auth_session.proof_expired_time < datetime.now() and auth_session.proof_status == AuthSessionState.NOT_STARTED:
+        logger.info("PROOF EXPIRED")
+        auth_session.proof_status = AuthSessionState.EXPIRED
+        await AuthSessionCRUD(db).patch(
+            str(auth_session.id), AuthSessionPatch(**auth_session.dict())
+        )
```

Comment:
This looks extremely similar to the code in `acapy_handler.py`: any chance we can consolidate it/not have this duplication? If the duplication is needed, could you elaborate on why?

Reply:
Yeah... I saw this possible duplication earlier. I just didn't know what sort of voodoo magic the older logic was doing. Thought it was some kind of keycloak thing so I didn't want to touch it.
If you're sure this is a duplication, I can consolidate.

Reply:
It looks like the code is basically doing teh same steps - at least in part. I'd assess whether it makes sense to consolidate rather than have them duplicated, that's all

Reply:
I'm on the same page. I'd rather not have duplication.